synapse/storage/databases/main/events_worker.py

```diff
@@ -1400,7 +1448,10 @@ def get_all_new_backfill_event_rows(txn):
                 " LIMIT ?"
             )
             txn.execute(sql, (-last_id, -current_id, instance_name, limit))
-            new_event_updates = [(row[0], row[1:]) for row in txn]
+            new_event_updates: List[Tuple[int, Tuple[str, str, str, str, str, str]]] = [
+                cast(Tuple[int, Tuple[str, str, str, str, str, str]], (row[0], row[1:]))
+                for row in txn
+            ]
```

Comment:
Feel like I've come across this problem recently. See https://github.com/matrix-org/synapse/pull/11354/files#r752485724

Reply:
Ahh sorry, I wasn't necessarily advocating to do follow my style---just thinking out loud! I'm happy either way.